localstack/services/sns/provider.py

```diff
@@ -753,6 +596,7 @@ def subscribe(
                 )
 
         store.subscriptions[subscription_arn] = subscription
+        store.topic_subscriptions[topic_arn] = store.topic_subscriptions.get(topic_arn) or []
         store.topic_subscriptions[topic_arn].append(subscription_arn)
```

Comment:
```suggestion
        store.topic_subscriptions.setdefault(topic_arn, []).append(subscription_arn)
        # or maybe
        topic_subscriptions = store.topic_subscriptions.setdefault(topic_arn, [])
        topic_subscriptions.append(subscription_arn)
```
nit: maybe a personal feeling but I like using `setdefault` instead of `or` in a `defaultdict` fashion

Reply:
Nice, TIL about `dict.setdefault` :+1: 

Added in 321c4c32